aodncore/util/misc.py

```diff
@@ -95,6 +96,22 @@ def format_exception(exception):
     return "{cls}: {message}".format(cls=exception.__class__.__name__, message=exception)
 
 
+def get_pattern_subgroups_from_string(str, pattern):
```

Comment:
Why _sub_groups? I think `get_pattern_groups_from_string` would be fine, or even `get_pattern_dict_from_string`.

Reply:
we just changed that with Leigh, but we can have an un-official meeting with @lwgordonimos ?

Reply:
It's the terminology used by the re library:
https://docs.python.org/2/library/re.html#re.MatchObject.groupdict

The primary use case is "fields from an IMOS filename", but this is really a generic function to get any "named subgroups" from a "string", so I was thinking to not imply that it is coupled tightly to that use case.

If there's a need for a wrapper specifically for *fields* it could also do validation against http://content.aodn.org.au/Documents/IMOS/Conventions/IMOS_NetCDF_Conventions.pdf it could be named in line with those conventions?

I'm not overly precious about it, but that was the thinking anyway!